inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <tuple>
+
+#include "transformations/insert_transpose_before_matmul.hpp"
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <transformations/init_node_info.hpp>
+
+namespace testing {
+
+namespace {
+
+std::shared_ptr<ngraph::Function> createFunction(const ngraph::PartialShape& input_values,
+                                                     const ngraph::Shape& reshape_values,
+                                                     const ngraph::Shape& matmul_values) {
+    auto input_params = std::make_shared<ngraph::opset7::Parameter>(ngraph::element::i64, input_values);
+
+    auto new_shape = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{reshape_values.size()}, reshape_values);
+    auto reshape_operation = std::make_shared<ngraph::opset7::Reshape>(input_params, new_shape, true);
+
+    auto constant = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{matmul_values.size()}, matmul_values);
+    auto matmul_operation = std::make_shared<ngraph::opset7::MatMul>(reshape_operation, constant);
+
+    auto result = std::make_shared<ngraph::opset7::Result>(matmul_operation);
+    return std::make_shared<ngraph::Function>(ngraph::ResultVector{result},
+                                              ngraph::ParameterVector{input_params});
+}
+
+TEST(TransformationTests, InsertTransposeBeforeMatmulTestShapeNotSupported) {
+    std::shared_ptr<ngraph::Function> func(nullptr), reference_func(nullptr);
+
+    {
+        func = createFunction(ngraph::PartialShape{2, 9}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+        ngraph::pass::Manager m;
+        m.register_pass<ngraph::pass::InitNodeInfo>();
+        m.register_pass<GNAPluginNS::InsertTransposeBeforeMatmul>();
+        m.run_passes(func);
+        ASSERT_NO_THROW(check_rt_info(func));
+    }
+
+    reference_func = createFunction(ngraph::PartialShape{2, 9}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(func, reference_func);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST(TransformationTests, InsertTransposeBeforeMatmulTestReshapeInOutEq) {
```

Comment:
Looks like we can use TEST_P for negative cases here and do not duplicate the test logic twice.